components/PokemonListItem/index.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useMemo } from "react";
+import Link from "next/link";
+import styles from "./pokemonListItem.module.css";
+import Image from "next/image";
+
+type PokemonListItemProps = {
+  pokemon: Record<string, any>;
+};
+
+export default function PokemonListItem({
+  pokemon,
+}: PokemonListItemProps): React.ReactElement {
+  const urlArr = pokemon?.url.split("/");
+  const pokeIndex = useMemo(() => {
```

Comment:
I don't know what's happening here because I don't know the structure of how `pokemon` looks like - hence we need a more defined type
Anywho, 
I like that you've memoized but it seems like premature optimisation since this `pokeIndex` const is not subscribed to in any other hook OR doesn't look like a very expensive calculation so we can avoid it and replace it with like a `getPokeIndex(pokemon)` function util

Reply:
Done as stated